classes/ticket.js

```diff
@@ -0,0 +1,200 @@
+const { Collection } = require("discord.js");
+const Discord = require("discord.js");
+const discordServices = require('../discord-services');
+
+class Ticket {
+    constructor(guild, question, caveOptions, requester, hackers, number, ticketMsg, inactivePeriod, bufferTime) {
+        this.guild = guild;
+        this.category;
+        this.question = question;
+        this.text;
+        this.voice;
+        this.caveOptions = caveOptions;
+        this.requester = requester;
+        this.hackers = hackers;
+        this.hackercount = hackers.size + 1;
+        this.mentors = [];
+        this.number = number;
+        this.ticketMsg = ticketMsg;
+        this.inactivePeriod = inactivePeriod;
+        this.bufferTime = bufferTime;
+        this.interval;
+        this.deletionSequence = false;
+        this.excluded = false;
+        this.init();
+    }
+    async includeExclude(exclude) {
+        this.excluded = exclude;
+        if (!exclude) {
+            this.createActivityListener();
+        }
+    }
+
+    async createCategory() {
+        this.category = await this.guild.channels.create("Ticket - " + this.number, {
+            type: 'category',
+            permissionOverwrites: [
+                {
+                    id: discordServices.everyoneRole,
+                    deny: ['VIEW_CHANNEL'],
+                }
+            ]
+        });
+
+        this.text = await this.guild.channels.create('banter', {
+            type: 'text',
+            parent: this.category
+        });
+
+        this.voice = await this.guild.channels.create('discussion', {
+            type: 'voice',
+            parent: this.category
+        });
+    }
+
+    async init() {
+
+        /**
+         * @type {Discord.Collection<Discord.Snowflake, Discord.GuildEmoji>} - <guild emoji snowflake, guild emoji>
+         */
+        const ticketEmojis = new Discord.Collection();
+        ticketEmojis.set(this.caveOptions.giveHelpEmoji.name, this.caveOptions.giveHelpEmoji);
+
+        // the embed used to inform hackers and users of the open ticket, sent to the ticket text channel
+        const openTicketEmbed = new Discord.MessageEmbed()
```

Comment:
Since this embed will always stay constant for all tickets, we can make it a class variable. Same for the openTicketEmbedMsg, since there is only one per ticket, we could make it a class variable.